### PR TITLE
chore: update rust toolchain

### PR DESCRIPTION
**Pull request by hansieodendaal**
Description
---
- Updated rust toolchain to be compatible with the latest `tari-crypto`.
- Pointed to `tari-crypto/main` on GitHub while waiting for the updated crate to be released - this can be changed back later.
- Most changes related to:
  - a reference being passed into a function that is immediately being;
    dereferenced by the compiler
  - enum naming conventions;
  - superfluous if let bindings.

**Note:** A compiler version was chosen that had all the boxes ticked for the three major operating systems and that would still allow avx2, thus no issues with `packed_simd_2`. These versions were tested:
- `nightly-2021-09-28`: issues with `packed_simd_2`
- `nightly-2021-09-15`: issues with `packed_simd_2`
- _`nightly-2021-08-17`: **the chosen version**, worked fine_
- `nightly-2021-06-01`: this version worked fine
- `nightly-2021-05-09`: the original version, does not work due to changes in `subtle-ng` via `tari-crypto`

Motivation and Context
---
See above.

How Has This Been Tested?
---
Fully compiled with clippy, unit tests succeeded.

### CODE REVIEW

**Comment by delta1** [comment]
Will test locally - but please update the nightly version required in the github actions `ci.yml` file

**Comment by delta1** [comment]
There are many format warnings/errors - you'll need to run `cargo fmt --all` 

Also no need to amend/force push any more, it's easier to review with additional commits (since they are now squashed on merge)

**Comment by hansieodendaal** [comment]
Thanks @delta1, updated the toolchain everywhere. The last-minute formatting caught me out, so that is also done.

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 version: 2.1
 
 defaults:
-  rust_image: &rust_image quay.io/tarilabs/rust_tari-build-with-deps:nightly-2021-05-09
+  rust_image: &rust_image quay.io/tarilabs/rust_tari-build-with-deps:nightly-2021-08-17
```
Comment (delta1) [inline .circleci/config.yml:4]: I will build and push this version to quay.io as it doesn't exist yet

Comment (hansieodendaal) [reply .circleci/config.yml:4]: Ah... the problem with the integration tests. Thanks.

**Review by delta1 — APPROVED** [review]

**Review by stringhandler — APPROVED** [review]

**Pull request merged**